signer.js

```diff
@@ -1,32 +1,96 @@
 'use strict'
 
-const cookieSignature = require('cookie-signature')
+const crypto = require('crypto')
 
-module.exports = function (secret) {
-  const secrets = Array.isArray(secret) ? secret : [secret]
-  const [signingKey] = secrets
+function SignerFactory (secrets, algorithm = 'sha256') {
+  if (!(this instanceof SignerFactory)) {
+    return new SignerFactory(secrets, algorithm)
+  }
```

Comment:
This is incorrect. In JavaScript land, this is no longer a "factory" function but a "class function". So the export would either be `function Signer(...) {...}` or the original factory would return a new instance of `Signer`.

My original feedback was asking for:

```js
module.exports = function () {
  // do some stuff

  return { sign, unsign }

  function sign() {
    // sign code
  }

  function unsign() {
    // unsign code
  }
}
```

I'm not opposed have a `Signer` "class" that the exported factory function returns. But we set that up and name things correctly.

Reply:
Ah. Allrighty. Will do

Reply:
Yes.

Reply:
renamed SignerFactory to Signer. Waiting for your feedback ;)